addons/website/controllers/model_page.py

```diff
@@ -0,0 +1,134 @@
+import ast
+
+import werkzeug
+
+from odoo.addons.http_routing.models.ir_http import slug, unslug
+from odoo.http import Controller, request, route
+from odoo.osv.expression import AND, OR
+
+
+class ModelPageController(Controller):
+    pager_step = 20
+
+    @route([
+        "/model/<string:page_name_slugified>",
+        "/model/<string:page_name_slugified>/page/<int:page_number>",
+        "/model/<string:page_name_slugified>/<string:record_slug>",
+    ], website=True, auth="public")
```

Comment:
Maybe irrelevant, just quickly commenting, but can't we use modelconverter somehow?
https://discord.com/channels/678381219515465750/687337739129323543/1139134167351820319

Reply:
not really, we don't know which model we are dealing with beforehand